src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs

```diff
@@ -941,6 +941,27 @@ internal override void Enter(EnterParams @params, int depth)
 #if !UNO_HAS_BORDER_VISUAL
 		UpdateBorder();
 #endif
+
+		// We do this in Enter not Loaded since Loaded is a lot more tricky
+		// (e.g. you can have Unloaded without Loaded, you can have multiple loaded events without unloaded in between, etc.)
+		if (!IsNativeHost)
+		{
+			TryRegisterNativeElement(null, Content);
+		}
+		if (IsNativeHost)
```

Comment:
```suggestion
		else
```

Reply:
Oh, I think the intent is that TryRegisterNativeElement may change IsNativeHost. It's worth a comment if that's the case

Reply:
No, if `!IsNativeHost`, then try to register as a native host (which might make `IsNativeHost` true), and whether or not `IsNativeHost` was originally true or just became true after the `TryRegisterNativeElement` true, we want to attach the native element. It does look weird though, I will add a comment.